pkg/server/register.go

```diff
@@ -45,25 +47,41 @@ var (
 )
 
 func (i *InventoryServer) ServeHTTP(resp http.ResponseWriter, req *http.Request) {
+	logrus.Debugf("Incoming HTTP request for %s", req.URL.Path)
 	// get the machine registration relevant to this request
 	registration, err := i.getMachineRegistration(req)
 	if err != nil {
 		http.Error(resp, err.Error(), http.StatusNotFound)
 		return
 	}
 
-	// attempt to authenticate the machine, if the machine is nil, authentication has failed
-	inventory, w, err := i.authMachine(resp, req, registration.Namespace)
+	if !websocket.IsWebSocketUpgrade(req) {
```

Comment:
how does this works? we run the auth after this part? Not sure I understand it correctly

Reply:
Basically, in order to switch a plan HTTP connection to a websocket one, it is the client that must add an "Upgrade" header in the HTTP GET request first (the gorilla/websocket library Dial() function does this automatically for the client, so it is hidden in the code). The idea here is, as soon as we receive an HTTP GET, let's check if it contains the UPGRADE header to upgrade to a websocket connection. If so, it should be a register client, otherwise is just a request to download the registration yaml to feed the elemental-register cli (via curl or any plain HTTP client).